src/IO/CompressionMethod.h

```diff
@@ -4,7 +4,7 @@
 #include <string>
 
 #include <Core/Defines.h>
-
+#include <Core/Settings.h>
```

Comment:
Can use forward declaration of `Settings` instead, and maybe it is better to pass value of the setting explicitly? (like `buf_size`)